addon/types.ts

```diff
@@ -0,0 +1,36 @@
+import Mix from '@ember/polyfills/types';
+
+export type Nullable<T> = T | null | undefined;
+
+export interface PlainObject {
+  [key: string]: string | PlainObject | PlainObject[];
+}
+
+export interface PlainHeaders {
+  [key: string]: string | undefined | null;
+}
+
+export type Method =
+  | 'HEAD'
+  | 'GET'
+  | 'POST'
+  | 'PUT'
+  | 'PATCH'
+  | 'DELETE'
+  | 'OPTIONS';
+
+export type AjaxOptions = {
+  url: string;
+  type: Method;
+  data?: PlainObject | BodyInit;
+  headers?: PlainHeaders;
+};
+
+export type FetchOptions = Mix<
+  AjaxOptions,
+  { body?: BodyInit | null; method?: Method }
+>;
+
+export function isPlainObject(obj: any): obj is PlainObject {
+  return Object.prototype.toString.call(obj) === '[object Object]';
```

Comment:
this isn't really the right implementation for this guard, since it will pass for lots of objects that aren't `PlainObject`. For example `{ foo() { return 'bar'; } }` will still coerce to string as `[object Object]`

Reply:
I want to preserve all current behaviors so it be a type only PR. (I assume `Object.prototype.toString.call(obj)` equals `String(obj)`. There exists ways to break this but we can leave it in another fix PR.